### PR TITLE
fix(http): fix template not found error

### PR DESCRIPTION
**Pull request by kelwang**
Closes https://github.com/influxdata/influxdb/issues/12864
Closes https://github.com/influxdata/influxdb/issues/12871

add an error check for not found errors for documents

### CODE REVIEW

File: kv/document.go [hunk 1]
```diff
@@ -81,6 +81,12 @@ func (s *Service) FindDocumentStore(ctx context.Context, ns string) (influxdb.Do
 	})
 
 	if err != nil {
+		if IsNotFound(err) {
```
Comment (desa) [inline kv/document.go]: I don't think this solves #12864. This function is just used to lookup a namespace not the actual document.

Comment (kelwang) [reply kv/document.go]: Thanks a lot

File: kv/document.go [hunk 2]
```diff
@@ -582,6 +582,12 @@ func (s *DocumentStore) FindDocuments(ctx context.Context, opts ...influxdb.Docu
...
```
Comment (desa) [inline kv/document.go]: What do you think about un-nesting the conditional so it's something like

```go
if IsNotFoundErr(err) {
   ...
}

if err != nil {
...
}
```

**Comment by kelwang** [comment]
> Could you add a test for this?

i added another ticket for the testing, https://github.com/influxdata/influxdb/issues/12871

**Comment by desa** [comment]
there are tests, they just aren't unit tests, it should be possible to extend them.

https://github.com/influxdata/influxdb/blob/6c6247efb3a45558be2a211c81dff8c3bae9fe75/testing/document.go#L15

**Comment by kelwang** [comment]
> there are tests, they just aren't unit tests, it should be possible to extend them.
> 
> [influxdb/testing/document.go](https://github.com/influxdata/influxdb/blob/6c6247efb3a45558be2a211c81dff8c3bae9fe75/testing/document.go#L15)
> 
> Line 15 in [6c6247e](/influxdata/influxdb/commit/6c6247efb3a45558be2a211c81dff8c3bae9fe75)
> 
>  func NewDocumentIntegrationTest(store kv.Store) func(t *testing.T) {

added

**Pull request merged**